src/containers/NahmiiWithdraw/index.js

```diff
@@ -0,0 +1,823 @@
+    const nahmiiStagedBalanceAfter = {
+      amount: nahmiiStagedBalanceAfterAmt,
+      usdValue: nahmiiStagedBalanceAfterAmt.times(assetToWithdrawUsdValue),
+    };
+    const baseLayerBalAfterAmt = baseLayerBalanceBefore.amount.minus(amountToWithdraw);
+    const baseLayerBalanceAfter = {
+      amount: baseLayerBalAfterAmt,
+      usdValue: baseLayerBalAfterAmt.times(assetToWithdrawUsdValue),
+    };
+
+    // constuct ether before and after balances
+    const baseLayerEthBalanceBefore = {
+      amount: baseLayerEthBalance.balance,
+      usdValue: baseLayerEthBalance.balance.times(ethUsdValue),
+    };
+    const baseLayerEthBalanceAfterAmount = assetToWithdraw.symbol === 'ETH'
+        ? baseLayerEthBalanceBefore.amount.plus(amountToWithdraw).minus(transactionFee.amount)
+        : baseLayerEthBalanceBefore.amount.minus(transactionFee.amount);
+    const baseLayerEthBalanceAfter = {
+      amount: baseLayerEthBalanceAfterAmount,
+      usdValue: baseLayerEthBalanceAfterAmount.times(ethUsdValue),
+    };
+
+    const maxExpirationTime = ongoingChallenges.get('details').reduce((max, challenge) => {
+      const { expirationTime } = challenge;
+      return max > expirationTime ? max : expirationTime;
+    }, 0);
+    const totalStagingAmountBN = ongoingChallenges.get('details').reduce((sum, challenge) => {
+      const { intendedStageAmount } = challenge;
+      const { amount } = intendedStageAmount.toJSON();
+      return sum.plus(new BigNumber(amount));
+    }, new BigNumber(0));
+    const totalStagingAmount = totalStagingAmountBN.div(new BigNumber(10).pow(assetToWithdrawMaxDecimals));
+
+    const stagedAsset = nahmiiStagedAssets.find((a) => a.symbol === assetToWithdraw.symbol) || { balance: new BigNumber(0) };
+    const requiredSettlementAmount = this.getRequiredSettlementAmount(stagedAsset.balance, amountToWithdraw);
+    const nahmiiBalanceAfterStagingAmt = nahmiiBalanceBeforeAmt.minus(requiredSettlementAmount);
+    const nahmiiBalanceAfterStaging = {
+      amount: nahmiiBalanceAfterStagingAmt,
+      usdValue: nahmiiBalanceAfterStagingAmt.times(assetToWithdrawUsdValue),
+    };
+    const totalSettleableStageAmountBN = settleableChallenges.get('details').reduce((sum, challenge) => {
+      const { amount } = challenge.intendedStageAmount.toJSON();
+      const intendedStageAmount = new BigNumber(amount);
+      return sum.plus(intendedStageAmount);
+    }, new BigNumber(0));
+
+    const totalSettleableStageAmount = totalSettleableStageAmountBN.div(new BigNumber(10).pow(assetToWithdrawMaxDecimals));
+
+    const walletType = currentWalletWithInfo.get('type');
+    const disableWithdrawButton =
+      amountToWithdraw.toNumber() <= 0 ||
+      baseLayerBalAfterAmt.isNegative() ||
+      baseLayerEthBalanceAfterAmount.isNegative() ||
+      !walletReady(walletType, ledgerNanoSInfo, trezorInfo);
+
+    const disableSettleButton = nahmiiBalanceAfterStagingAmt.isNegative() || !walletReady(walletType, ledgerNanoSInfo, trezorInfo);
```

Comment:
Disable if `baseLayerBalAfter` is <0? Or the error message shown if insufficient funds could be improved